pkg/cloudinit/cloudinit.go

```diff
@@ -67,6 +67,12 @@ func WriteFiles(paths ...string) error {
 		return err
 	}
 
+	// ensure directory which holds files exists before attempting write
+	err = os.MkdirAll(fileDir, os.ModeDir)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Since we might need to save other files from cloud-init in the future, how about instead:

        fileDir := filepath.Dir(p)
	err = os.MkdirAll(fileDir, os.ModeDir)
	if err != nil {
		return err
	}

in the loop `for _, p := range paths {`?

Reply:
done, sorry about that i misunderstood the original ask.